File: Threshold/Otsu/otsu.py
import cv2 as cv
import numpy as np 
from matplotlib import pyplot as plt

input_image = input('Enter the image: ')
img = cv.imread(input_image, 0)

ret1, th1 = cv.threshold(img, 0, 255, cv.THRESH_BINARY+cv.THRESH_OTSU)

cv.imshow("Otsu", th1)
cv.waitKey(0)

File: Threshold/Otsu/otsu.cpp
#include "opencv2/imgproc.hpp"
#include "opencv2/imgcodecs.hpp"
#include "opencv2/highgui.hpp"
#include <iostream>
using namespace cv;
using std::cout;

int main(int argc, const char** argv)
{
    Mat input_image = imread(argv[1], IMREAD_GRAYSCALE);
    Mat output;
    threshold(input_image, output, 0, 255, THRESH_OTSU);
    imshow("Original", input_image);
    imshow("Otsu", output);
    waitKey(0);
}
